agent-identity/python/ciba-on-behalf-of-flow/sleeping-guardian/aurelius_agent_llm.py

```
@@ -0,0 +1,497 @@
+                "threshold": self.price_threshold,
+                "budget": self.trade_amount,
+                "scopes": ", ".join(self.current_scopes)
+            })
+
+            output = response.get("output", "")
+            print(f"[Aurelius AI] ✓ Trade result: {output}")
+
+            # Check if the LLM refused to call the tool or reported a scope error
+            scope_refusal_keywords = ["insufficient_scope", "stock:trade", "scope", "authorization", "permission"]
+            if any(kw in output.lower() for kw in scope_refusal_keywords) and "success" not in output.lower():
+                print(f"[Aurelius AI] 🔐 Trade blocked due to insufficient scope — triggering CIBA")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)